torch/testing/_internal/common_device_type.py

```diff
@@ -205,8 +213,63 @@ def _get_precision_override(self, test, dtype):
     # Creates device-specific tests.
     @classmethod
     def instantiate_test(cls, name, test):
-        test_name = name + "_" + cls.device_type
+        # Handles tests using the ops decorator
+        if hasattr(test, "op_list"):
+            for op in test.op_list:
+                base_op_test_name = name + "_" + op.name + "_" + cls.device_type
+
+                # Acquires dtypes, using the op data if unspecified
+                dtypes = cls._get_dtypes(test)
+                if dtypes is None:
+                    if cls.device_type == 'cpu' and op.dtypesIfCPU is not None:
+                        dtypes = op.dtypesIfCPU
+                    elif (cls.device_type == 'cuda' and not TEST_WITH_ROCM
+                          and op.dtypesIfCUDA is not None):
+                        dtypes = op.dtypesIfCUDA
+                    elif (cls.device_type == 'cuda' and TEST_WITH_ROCM
+                          and op.dtypesIfROCM is not None):
+                        dtypes = op.dtypesIfROCM
+                    else:
+                        dtypes = op.dtypes
+
+                # Inverts dtypes if the function wants unsupported dtypes
+                if test.unsupported_dtypes_only is True:
+                    dtypes = [d for d in get_all_dtypes() if d not in dtypes]
+
+                for dtype in dtypes:
+                    # Constructs dtype suffix
+                    if isinstance(dtype, (list, tuple)):
+                        dtype_str = ""
+                        for d in dtype:
+                            dtype_str += "_" + str(d).split('.')[1]
```

Comment:
Can't wait for the day when someone gets the cute idea of `torch.quantized.qint8` and then this doesn't work right :) At least the assert will fire if this happens...

Reply:
Ug... "does not support quantized dtypes" is right